mesh_segmenter/src/convex_hull.cpp

```diff
@@ -0,0 +1,83 @@
+#include <mesh_segmenter/convex_hull.h>
+
+ConvexHullGenerator::ConvexHullGenerator(){}  
+
+void ConvexHullGenerator::makeMesh(const std::string& input, pcl::PointCloud<pcl::PointXYZ>& inMesh)
+{
+  //expects a .ply file
+  pcl::PLYReader Reader;
+  Reader.read(input, inMesh); //populate inMesh
+  return;
+}
+
+
+void ConvexHullGenerator::cleanMesh(const pcl::PointCloud<pcl::PointXYZ>& outMesh, pcl::PolygonMesh& outMeshPoly)
+{
+  //find centroid coords by finding average x, y, z 
+
+  Eigen::Matrix< float, 4, 1 > mid;
+  int centroid_success = pcl::compute3DCentroid(outMesh, mid);
+  Eigen::Vector3d midVec= {mid[0], mid[1], mid[2]};
+
+  //invert bad polygons --------------------------------
+  for (int t=0; t < (outMeshPoly.polygons.size()); t++)
+  {
+    pcl::Vertices verts;
+    verts = outMeshPoly.polygons[t];
+
+    pcl::PointXYZ a = outMesh.points[verts.vertices[0]];
+    pcl::PointXYZ b = outMesh.points[verts.vertices[1]];
+    pcl::PointXYZ c = outMesh.points[verts.vertices[2]];
+
+    Eigen::Vector3d p0 = {a.x, a.y, a.z};
+    Eigen::Vector3d p1 = {b.x, b.y, b.z};
+    Eigen::Vector3d p2 = {c.x, c.y, c.z};
+
+    Eigen::Vector3d v1 = p1 - p0; 
+    Eigen::Vector3d v2 = p2 - p0;
+    Eigen::Vector3d d = p0 - midVec;
+    Eigen::Vector3d normal = v1.cross(v2);
+    float works = d.dot(normal);
```

Comment:
variable name is not helpful. you are computing a projection of one vector onto another. When that is negative, it implies they are aligned in opposing direction. Perhaps that might be a better name.